### PR TITLE
make preset applicability wording a bit more obvious

### PR DESCRIPTION
**Pull request by johnny-bit**
Judging by problem in 9339 I think it might've been caused by a correct but not "obvious" wording

this PR changes wording:

- normal images -> non-raw
- raw -> raw

should make it clear what's what.

### CODE REVIEW

**Review by elstoc — APPROVED** [review]
much clearer, thanks

**Comment by elstoc** [comment]
Thinking again... Are normal/raw/HDR mutually exclusive? If so, then HDR is also non-raw so maybe we need a better term?

**Comment by johnny-bit** [comment]
It's a bit different

So:
 - non-raw images are already demosaiced (check dt_is_rawprepare_supported)
 - raw files are non-demosaiced (double negation... bleh. mosaiced) (check dt_is_ldr
 - HDRs CAN be raws AND can be non-raws

as for color:
 - raw / non-raw can be monochrome
 - raw / non-raw can be colour
 - raw / non-raw can't be both
 - raw / non-raw can't be neither

So the "full correctness" could be:

### format
 - non-raw images
 - raw files
### HDR
 - HDRs
### Color
 - monochrome
 - color

What do you think? I'd rather not go that way (mabe add it in the manual)

**Comment by elstoc** [comment]
Ok makes sense. We'll put a clarification in the user manual. I'll raise a dtdocs issue for now.

**Comment by johnny-bit** [comment]
Further correction: HDRs MUST be either RAW or NON-RAW... so i wonder if it'd be possible to specify auto preset that is not applied to raws or non-raws but is applied to HDRs... and to dat say just to non-raw hdrs... but that's complicating life imo ;)

File: src/gui/presets.c [hunk 1]
```diff
@@ -55,7 +55,7 @@ static const char *_dt_gui_presets_aperture_value_str[]
 
 // format string and corresponding flag stored into the database
 static const char *_dt_gui_presets_format_value_str[5]
-    = { N_("normal images"), N_("raw"), N_("HDR"), N_("monochrome"), N_("color") };
+    = { N_("non-raw images"), N_("raw files"), N_("HDR"), N_("monochrome"), N_("color") };
```
Comment (TurboGit) [inline src/gui/presets.c:58]: why not "raw images" to be consistent with "non-raw images" ?

Comment (elstoc) [reply src/gui/presets.c:58]: Raw files are not images.

Comment (TurboGit) [reply src/gui/presets.c:58]: Indeed !

Comment (TurboGit) [reply src/gui/presets.c:58]: So maybe just:  "images" and "raw files" ?

Comment (johnny-bit) [reply src/gui/presets.c:58]: this is deliberate double impact on my side.

all images are non-raw. that's why "normal images" was perfectly valid name... but people got confused so "non-raw images"
raws are essentially sensor data-dumps + metadata. And without processing the sensor dump is hardly viewable as image so it's a file then.
It'd be great if we worked towards consistency to help user mental load with that: raws aren't images and images aren't raws and you can get image from raw but can't get raw from image. So ultimately in some future ver the wording could be simplified to "images" & "raws" but we're not at that point yet.

Comment (johnny-bit) [reply src/gui/presets.c:58]: @elstoc gave me good idea. lemme test it with you.

Comment (johnny-bit) [reply src/gui/presets.c]: removed redundancy. now we have it super obvious: "non-raw" and "raw". That should clear ALL confusion :) and let's hope I'm wrong about double impact being needed :)

**Comment by AlicVB** [comment]
Side note (to be seen as an extension of this PR) : 
> ### format
> 
>     * non-raw images
> 
>     * raw files
> 
> 
> ### HDR
> 
>     * HDRs
> 
> 
> ### Color
> 
>     * monochrome
> 
>     * color

Maybe I'm wrong, but if we want to be more clear for the user, imho, we can come with a better UI design that would group the widgets as you have done here. An example with combobox : 
- format [all - non-raw only - raw only]
- HDR [yes - no] <= no sure exactly, there must be a "whatever" entry too... (I've never deal with HDRs)
- color [all - monochrome only - color only]

We can also keep checkboxes with some grouping (on same line for example) and some code to avoid the "no selection" where needed...

All that is just gui code, I can take care in another PR if you want

**Comment by johnny-bit** [comment]
@AlicVB - I like your "combo boxes" proposal, since it would essentially prevent "misunderstanding" like the one in https://github.com/darktable-org/darktable/issues/9339#issuecomment-869048808

the only thing I don't understand deeply enough is HDR flag - I need to test it's conditionality in applying.

**Comment by AlicVB** [comment]
> the only thing I don't understand deeply enough is HDR flag - I need to test it's conditionality in applying.

same as me. I'll wait for your finding then :)

**Comment by johnny-bit** [comment]
@AlicVB - I think I got the picture with HDR flag and it's... Well:

- File is "RAW HDR", preset has selected "normal+raw" and deselected HDR - preset gets applied
- File is "normal HDR", preset has selected "normal+raw" and deselected HDR - preset gets applied
- File is "RAW HDR", preset has selected JUST HDR - prest gets applied
- File is "normal HDR", preset has selected JUST HDR - prest gets applied
- File is "RAW HDR", preset has JUST RAW selected - preset gets applied
- File is "normal HDR", preset has JUST RAW selected - preset does not get applied
- File is "RAW HDR", preset has JUST normal selected - preset does not gets applied
- File is "normal HDR", preset has JUST normal selected - preset gets applied

So essentially - HDR flag allows to setup preset that applies to just HDRs. There's currently no way to auto apply preset in a way that would not apply to HDRs because either way HDR files MUST be either normal images or RAWs

There's another fun: Color flags :)

- if preset has deselected normal, raw & HDR but selected mono+color: gets applied to ALL
- If preset has selected normal, raw & HDR but deselected mono+color: doesn't get applied at all


This was my exploration into current preset logic.

Hope this clears everything up and you can make something out of it :)

**Comment by AlicVB** [comment]
ouch... quite complex, indeed ! I will try to find where all this tests are in the code.
By any chance, do you have some link for hdr (normal and raw) files ? That would save me some search time ;)
Thanks for your tests !

**Comment by johnny-bit** [comment]
I got "normal" hdrs from hdrihaven, i've made raw hdr from 3 raws i had (and clicked create hdr in lighttable), monochrome raw i got from raw.pixls.us leica monochrom), raws i have aplenty, normal images were generated via export and monochrome normal images via gimp saving image in mono type :)

**Comment by johnny-bit** [comment]
Or rather - here's my sample set that has normal, raw, mono, color, raw hdr and normal hdr:) https://drive.google.com/drive/folders/1YQQSBvUh0dj-rItv_n9k5UiXwj0_wxw1?usp=sharing all images are CC0 or free.

**Review by TurboGit — APPROVED** [review]
This simple fix seems clearer to me, no better proposal so let's merging this now. Thanks.

**Comment by johnny-bit** [comment]
OK, but I'm still stoked for whatever @AlicVB will come up with :)

**Pull request merged**